src/julia-syntax.scm

```diff
@@ -4443,6 +4469,14 @@ f(x) = yt(x)
          (list-set e 3 (compile-body (cadddr e)
                                      (append (car (caddr e))
                                              (cadr (caddr e))) e)))
+        ((eq? (car e) 'foreigncall)
```

Comment:
This is handled by `compile-body`.

Reply:
Good point. I forgot everything ends up wrapped in a lambda, so it is impossible for this pass to end up encountering `foreigncall`.